### ISSUE TITLE
Make the compiler package part of the stdlib

### ISSUE THREAD

**Issue by cryzed** [nim-lang/Nim#3625]
It's required for all tools that need to have deeper knowledge about Nim's internal workings, such as NimSuggest. Currently it's shipped with the Windows installer but not included in the stdlib -- doing so would make sense. I'm not sure how exactly NimSuggest works, but I think this would also avoid scenarios where such tools are compiled with an older version of Nim while using the compiler package from a newer version -- possibly leading to incorrect autocompletion suggestions etc.

**Comment by dom96** [comment]
CC @Araq

**Comment by Araq** [comment]
@dom96 yeah ... in the works ... I prefer to make the compiler API more usable first though.

**Comment by dom96** [comment]
np.
